main.cpp

```diff
@@ -81,53 +87,56 @@ struct <#structName1#>                                //4
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float x { 0 }, y { 0 };
+    float multiply(float* updatedValue)      //12
     {
-        
+        std::cout << "U's x value: " << this->x << std::endl;
+        this->x = *updatedValue;
+        std::cout << "U's x updated value: " << this->x << std::endl;
+        while( std::abs(this->y - this->x) > 0.001f )
+        {
+            /*
+             write something that makes the distance between that->x and that->y get smaller
+             */
+            this->y += 0.1f;
+        }
+        std::cout << "U's y updated value: " << this->y << std::endl;
+        return this->y * this->x;
     }
 };
 
-struct <#structname2#>
+struct Z
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float multiply(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's x value: " << that->x << std::endl;
+        that->x = *updatedValue;
+        std::cout << "U's x updated value: " << that->x << std::endl;
```

Comment:
What must we always do before we use a pointer?
- [ ] revise

Reply:
check for nullptr